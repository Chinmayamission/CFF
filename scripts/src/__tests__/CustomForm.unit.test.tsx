import React from "react";
import { shallow, mount, render } from 'enzyme';
import CustomForm from "../form/CustomForm";
import sinon from "sinon";

it('renders default form without payment', () => {
  let schema = {
    "title": "Form",
    "type": "object",
    "properties": {
      "name": {
        "type": "string"
      }
    }
  };
  let uiSchema = {
    "name": {"ui:placeholder": "Name"}
  }
  const wrapper = render(
    <CustomForm schema={schema} uiSchema={uiSchema} />
  );
  expect(wrapper).toMatchSnapshot();
}); 

it('renders title and description with html', () => {
  let schema = {
    "title": "Form <img src='test'>",
    "description": "Hello <b>123</b><script>document.write('bad')</script><h1>Description</h1>",
    "type": "object",
    "properties": {
      "name": {
        "type": "string"
      }
    }
  };
  let uiSchema = {
    "title": "Yeah this is a form!"
  }
  const wrapper = render(
    <CustomForm schema={schema} uiSchema={uiSchema} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('submits form fail required validation', () => {
  let schema = {
    "title": "Form",
    "type": "object",
    "properties": {
      "name": {
        "type": "string"
      }
    },
    "required": ["name"]
  }; 
  let uiSchema = {
    "name": {"ui:placeholder": "Name"}
  }
  const spy = sinon.spy();
  const wrapper = mount(
    <CustomForm schema={schema} uiSchema={uiSchema} formData={{}} onSubmit={spy} />
  )
  let form = wrapper.find('form');
  form.simulate('submit');
  expect(form.text()).toContain("is a required field");
  expect(spy.calledOnce).toBe(false);
}); 

it('submits form success', () => {
  let schema = {
    "title": "Form",
    "type": "object",
    "properties": {
      "name": {
        "type": "string"
      }
    },
    "required": ["name"]
  }; 
  let uiSchema = {
    "name": {"ui:placeholder": "Name"}
  }
  const spy = sinon.spy();
  const wrapper = mount(
    <CustomForm schema={schema} uiSchema={uiSchema} formData={{"name": "Vishnu"}} onSubmit={spy} />
  );
  let form = wrapper.find('form');
  form.simulate('submit');
  expect(form.text()).not.toContain("is a required field");
  expect(spy.calledOnce).toBe(true);
});