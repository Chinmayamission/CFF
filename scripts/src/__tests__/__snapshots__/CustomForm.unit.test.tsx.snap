// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default form without payment 1`] = `
<div
  class="ccmt-cff-Page-FormPage"
>
  <form
    autocomplete="false"
    class="rjsf"
  >
    <div
      class="form-group field field-object"
    >
      <div
        class="container-fluid"
      >
        <h2
          class="ccmt-cff-form-title"
        >
          Form
        </h2>
        <div />
        <div
          class="row"
        >
          <div
            class="form-group field field-string"
          >
            <label
              class="control-label"
              for="root_name"
            >
              name
            </label>
            <input
              class="form-control"
              id="root_name"
              label="name"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <p>
      <button
        class="btn btn-info"
        type="submit"
      >
        Submit
      </button>
    </p>
  </form>
</div>
`;

exports[`renders title and description with html 1`] = `
<div
  class="ccmt-cff-Page-FormPage"
>
  <form
    autocomplete="false"
    class="rjsf"
  >
    <div
      class="form-group field field-object"
    >
      <div
        class="container-fluid"
      >
        <h2
          class="ccmt-cff-form-title"
        >
          Form 
          <img
            src="test"
          />
        </h2>
        <div>
          Hello 
          <b>
            123
          </b>
          <h1>
            Description
          </h1>
        </div>
        <div
          class="row"
        >
          <div
            class="form-group field field-string"
          >
            <label
              class="control-label"
              for="root_name"
            >
              name
            </label>
            <input
              class="form-control"
              id="root_name"
              label="name"
              placeholder=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <p>
      <button
        class="btn btn-info"
        type="submit"
      >
        Submit
      </button>
    </p>
  </form>
</div>
`;
